src/core/public/legacy/legacy_service.ts

```diff
@@ -96,25 +71,26 @@ export class LegacyPlatformService {
         linkToLastSubUrl: navLink.linkToLastSubUrl,
       })
     );
+  }
+
+  public start({ core, targetDomElement, plugins }: StartDeps) {
+    // Inject parts of the new platform into parts of the legacy platform
+    // so that legacy APIs/modules can mimic their new platform counterparts
+    require('ui/new_platform').__newPlatformStart__(core, plugins);
 
     // Load the bootstrap module before loading the legacy platform files so that
     // the bootstrap module can modify the environment a bit first
     this.bootstrapModule = this.loadBootstrapModule();
 
     // require the files that will tie into the legacy platform
     this.params.requireLegacyFiles();
```

Comment:
Moving `this.params.requireLegacyFiles()` into `start()` I assume will make `startCore` available at start of legacy platform.

Reply:
Yes! We may be able to make getting access to the new platform services even simpler with something like `import { coreStart } from 'ui/new_platform'` but I'd like to get this PR in first and make sure there aren't any corner cases where this won't work.